### ISSUE TITLE
Show overall progress bar when pulling/pushing

### ISSUE THREAD

**Issue by alberto56** [docker/docker#13186]
Especially for someone on a slow connection, the following output:

    docker push abc/xyz
    The push refers to a repository [abc/xyz] (len: 7)
    ae2eeb34f295: Image already exists 
    08159b0b0bd4: Image successfully pushed 
    83ec016e6d8c: Image successfully pushed 
    02ed4dc286c8: Image successfully pushed 
    5d626c56d596: Image successfully pushed 
    a793ca59282c: Pushing [===========================>                       ] 59.57 MB/108.6 MB
    
does not tell me how many images are left, so the progress bar is not meaningful. I would prefer to see:

    docker push abc/xyz
    The push refers to a repository [abc/xyz] (len: 7)
    ae2eeb34f295: Image already exists (1/60 images, 1/1009MB total)
    08159b0b0bd4: Image successfully pushed (2/60 images, 20/1009MB total)
    83ec016e6d8c: Image successfully pushed (3/60 images, 34/1009MB total)
    02ed4dc286c8: Image successfully pushed (4/60 images, 70/1009MB total)
    5d626c56d596: Image successfully pushed (5/60 images, 104/1009MB total)
    a793ca59282c: Pushing [===========================>                       ] 59.57 MB/108.6 MB

or perhaps as a percentage.

**Comment by stevenschlansker** [comment]
+1

**Comment by thaJeztah** [comment]
@alberto56 this looks to be a duplicate of https://github.com/docker/docker/issues/10894 (with the addition of mentioning "push")

Note that this feature might not be easy to implement, as there are several things that have to be taken into account when showing progress (the download itself is just one part of this, validating, extracting etc, is another)

Also have a look at https://github.com/docker/docker/issues/8385, for another issue (multiple clients pulling images in parallel)

**Comment by taraspogrebniak** [comment]
+1

**Comment by LK4D4** [comment]
Closing as duplicate of #10894
